packages/concerto-core/lib/decoratormanager.js

```diff
@@ -242,16 +297,88 @@ class DecoratorManager {
                 });
             }
         }
+    }
+
+    /**
+     * Adds decorator commands with index to the computed list
+     * @param {*} targetMap the target map to add the command to
+     * @param {targetKey} targetKey the target key to add the command to
+     * @param {DcsIndexWrapper[]} computedList the computed list to add the command to
+     * @private
+     */
+    static addToComputedList(targetMap, targetKey, computedList) {
+        const targetCommands = targetMap.get(targetKey);
+        if (targetCommands) {
+            targetCommands.forEach((dcsWithIndex) => {
+                computedList.push(dcsWithIndex);
+            });
+        }
+    }
+
+    /**
+     * Applies all the decorator commands from the DecoratorCommandSet to the ModelManager,
+     * this is optimized version of decorateModel method and will eventually replace its internal logic.
+     * @param {ModelManager} modelManager the input model manager
+     * @param {*} decoratorCommandSet the DecoratorCommandSet object
+     * @param {object} [options] - decorator models options
+     * @param {boolean} [options.validate] - validate that decorator command set is valid
+     * with respect to to decorator command set model
+     * @param {boolean} [options.validateCommands] - validate the decorator command set targets. Note that
+     * the validate option must also be true
+     * @param {boolean} [options.migrate] - migrate the decoratorCommandSet $class to match the dcs model version
```

Comment:
I see that you refactored the code above to use unbox these options. What are the risks of inconsistent representation of the same options?

Reply:
for `migrateAndValidate` method it doesn't feel right to me sending them as options as the internal functionality needs at least one of the flags to be passed for it to do something. And moreover its a private method that is used inside the `decorateModel` method.
There are similar couple of private methods that are truly dependent on a flag as the main argument for it to function,

for `executeCommand` I've sent the `enableDecoratorsForNamespace` as an option as its a public function and the flag is actually an optional flag.